mypy/checkpattern.py

```diff
@@ -202,7 +202,7 @@ def visit_value_pattern(self, o: ValuePattern) -> PatternType:
         typ = self.chk.expr_checker.accept(o.expr)
         typ = coerce_to_literal(typ)
         narrowed_type, rest_type = self.chk.conditional_types_with_intersection(
-            current_type, [get_type_range(typ)], o, default=current_type
+            current_type, [get_type_range(typ)], o, default=get_proper_type(typ)
```

Comment:
The issue was that on `mypy/checker.py` on `conditional_types` function, which is called by this `self.chk.conditional_types_with_intersection `, the `conditional_types` returns from here the default value:

https://github.com/python/mypy/blob/2037e4a068df6e1dcc8f76c37f53e04d62d64e80/mypy/checker.py#L7399

When on regular enum that has multiple choices the `conditional_types` returns here instead:

https://github.com/python/mypy/blob/2037e4a068df6e1dcc8f76c37f53e04d62d64e80/mypy/checker.py#L7415

But the first one returns wrong default value, so I am changing the default value here to be what the `get_proper_type` returns instead.